apps/alarm/request.js

```diff
@@ -17,48 +17,61 @@ function _createMd5 (extraData) {
 function request (params) {
   params.activity.get()
     .then(extraData => {
-      id++
-      var data = {
-        deviceId: extraData.deviceId,
-        deviceType: extraData.deviceTypeId,
-        intent: params.intent,
-        masterId: extraData.masterId,
-        nonce: extraData.deviceId + (new Date()).getTime() + id,
-        requestTimestamp: (new Date()).getTime(),
-        sessionId: (new Date()).getTime(),
-        sign: _createMd5(extraData),
-        signMethod: 'MD5',
-        businessParams: params.businessParams || {}
-      }
-      data = Buffer.from(JSON.stringify(data))
-      var req = https.request({
-        method: 'POST',
-        host: DEFAULT_HOST,
-        path: DEFAULT_URI || params.url,
-        headers: {
-          'Content-Type': 'application/json',
-          'Content-Length': data.length
+      try {
+        id++
+        var data = {
+          deviceId: extraData.deviceId,
+          deviceType: extraData.deviceTypeId,
+          intent: params.intent,
+          masterId: extraData.masterId,
+          nonce: extraData.deviceId + (new Date()).getTime() + id,
+          requestTimestamp: (new Date()).getTime(),
+          sessionId: (new Date()).getTime(),
+          sign: _createMd5(extraData),
+          signMethod: 'MD5',
+          businessParams: params.businessParams || {}
         }
-      }, (res) => {
-        var list = []
-        res.on('data', (chunk) => list.push(chunk))
-          .on('end', () => {
-            var result = Buffer.concat(list).toString()
-            if (res.statusCode !== 200) {
-              logger.error(`Error: failed get data with ${result}`)
-            } else {
-              if (typeof params.callback === 'function') {
-                params.callback(result)
+        data = Buffer.from(JSON.stringify(data))
+        var req = https.request({
+          method: 'POST',
+          host: DEFAULT_HOST,
+          path: DEFAULT_URI || params.url,
+          headers: {
+            'Content-Type': 'application/json',
+            'Content-Length': data.length
+          }
+        }, (res) => {
+          var list = []
+          res.on('data', (chunk) => list.push(chunk))
+            .on('end', () => {
+              var result = Buffer.concat(list).toString()
+              if (res.statusCode !== 200) {
+                logger.error(`Error: failed get data with ${result}`)
+                if (typeof params.callback === 'function') {
+                  params.callback(true, null)
+                }
+              } else {
+                if (typeof params.callback === 'function') {
+                  params.callback(false, result)
+                }
               }
-            }
-          })
-      })
+            })
+        })
 
-      req.on('error', (err) => {
+        req.on('error', (err) => {
+          logger.error(err && err.stack)
+          if (typeof params.callback === 'function') {
+            params.callback(true, null)
```

Comment:
The `param.callback` should be called only once, so we should add an internal state like `_called` to make sure that, @lanyuqingri :)